crates/bevy_render/src/render_resource/resource_macros.rs

```diff
@@ -120,16 +123,21 @@ macro_rules! render_resource_wrapper {
     };
 }
 
+pub trait ResourceId: Copy + Clone + Hash + Eq + PartialEq + Debug {
+    fn new() -> Self;
+    fn index(&self) -> usize;
+}
```

Comment:
A common trait shared by all resource ids.
This has to be imported wherever we call the new method (is this avoidable?).

Reply:
These Ids should probably be generational in the future.

Reply:
This trait is required to avoid duplicating code for the `PipelineCacheInternal` implementation.
There I create new ids via the `new` method and fetch the index using the `index` method.

Reply:
I could special case both `RenderPipelineId` and `CachePipelineId` from the rest of the render resource ids, but I do not think that this is the better solution.

Reply:
I have removed this in favor of a separate PipelineId trait.